elastic/elastic-agent
# Added skip audit/unenroll flag to uninstall command

- Enhancement

## What does this PR do?

Added a flag to skip fleet audit/unenroll while uninstalling elastic-agent. 

## Why is it important?

While uninstalling elastic-agent it tries to notify fleet server about the uninstall. But in somecases users might know that the fleet server is unreachable and this notification logs multiple failures continuously. Therefore having an option to skip this would enhance the end user experience.

## Checklist




## Disruptive User Impact

No disruptive user impact

## How to test this PR locally

- Setup ES and Kibana.
- Proceed with Fleetserver installation.
- Build the agent locally by running `SNAPSHOT=true PLATFORMS=linux/arm64 PACKAGES=tar.gz mage -v package`
- Extract the agent tar.
- Go to Kibana -> Fleet page -> Add Agent gives installation command. Use the same to install elastic-agent.
- After successful installation, Check in fleet page for the installed agent.
- Kill the fleet servers process.
- Try uninstalling elastic-agent using `sudo elastic-agent uninstall`
- Now as the fleet server is unreachable, the notify calls will fail. You will get the error logs like below
`[==  ] notify Fleet: network error: fail to notify audit/unenroll on fleet-server: all hosts failed: requester 0/1 to host https://192.168.1.10:8220/ errored: Post "https://192.168.1.10:8220/api/fleet/agents/a4888371-ef7b-4cc6-8df7-cc92b6c25990/audit/unenroll?": dial tcp 192.168.1.10:8220: connect:`
- Now restart the fleet server and install the elastic agent again.
- After successful installation, kill the fleet server.
- Now try uninstalling elastic-agent using `sudo elastic-agent uninstall --skip-fleet-audit`. The audit/unenroll call will be skipped and agent uninstall will happen without any errors.

## Related issues

- Closes #5757
